test/jdk/sun/util/resources/cldr/LikelySubtagLocalesTest.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2016, 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+ /*
+ * @test
+ * @bug 8145136 8202537 8221432 8251317 8258794 8265315 8306116 8346948
+ * @modules jdk.localedata
+ * @summary Tests LikelySubtags is correctly reflected in Locale.getAvailableLocales().
+ * @run junit/othervm -Djava.locale.providers=CLDR LikelySubtagLocalesTest
+ */
+import java.util.Arrays;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class LikelySubtagLocalesTest {
+
+    public static final List<Locale> AVAILABLE_LOCALES = Arrays.asList(Locale.getAvailableLocales());
+
+    /* Samples of locales that do not exist as .xml CLDR source files, but derived from
```

Comment:
New test looks good and is easier to understand. It was unclear to me why the previous test said "_Tests LikelySubtags is correctly reflected in Locale.getAvailableLocales()_" when it was testing all of the non-inferred locale tags.

Reply:
Thanks Justin. The test has been checking *all* the available locales which should contain likelySubtag inferred ones. While this ensures comprehensive testing, it is not practical to logically derive the expected locale sets, as it is not as simple as listing .xml files + likelySubtag inferred. Maintaining the locale set per CLDR release does not seem practical, which is why this change was made.